grails-app/views/product/_productSynonymsEdit.gsp

```diff
@@ -0,0 +1,127 @@
+<%@ page import="org.pih.warehouse.core.SynonymTypeCode" %>
+<g:form name="editSynonym" action="editProductSynonym" >
+    <g:hiddenField name="product.id" value="${product.id}" />
+    <g:hiddenField name="synonym.id" value="${synonym.id}" />
```

Comment:
as above - I'd consider adding null check here and in the below lines - `synonym?.locale?.language` etc